src/plugins/data/common/datasets/types.ts

```diff
@@ -32,6 +32,23 @@ export interface DataSourceMeta {
   supportsTimeFilter?: boolean;
 }
 
+export interface DataStructureCreatorProps<FetchOptions = unknown> {
+  /** list of selected data structures */
+  path: DataStructure[];
+  /** update list of selected data structures */
+  setPath: (newPath: DataStructure[]) => void;
+  /** level of current data structure */
+  index: number;
+  /** select the next data structure or dataset */
+  selectDataStructure: (item: DataStructure | undefined, newPath: DataStructure[]) => Promise<void>;
+  /** fetch data structure using fetch in type config */
+  fetchDataStructure: (
```

Comment:
how come this over `fetchNextDataStructure`?  the name for me indicated that if i pass a path then i would get the `next` data structure.

however if `fetchDataStructure` then i would be expect to get my `current` data structure